src/Admin/DevToolsUserAccess.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Class DevToolsUserAccess.
+ *
+ * @since 1.6.0
+ *
+ * @package AMP
+ */
+
+namespace AmpProject\AmpWP\Admin;
+
+use AmpProject\AmpWP\Infrastructure\Delayed;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use WP_Error;
+use WP_User;
+
+/**
+ * Class DevToolsUserAccess
+ *
+ * @since 1.6.0
+ */
+final class DevToolsUserAccess implements Delayed, Service, Registerable {
+
+	/**
+	 * User meta key enabling or disabling developer tools.
+	 *
+	 * @var string
+	 */
+	const USER_FIELD_DEVELOPER_TOOLS_ENABLED = 'amp_dev_tools_enabled';
+
+	/**
+	 * Get the action to use for registering the service.
+	 *
+	 * @return string Registration action to use.
+	 */
+	public static function get_registration_action() {
+		return 'rest_api_init';
+	}
+
+	/**
+	 * Runs on instantiation.
+	 *
+	 * @action rest_api_init
+	 */
+	public function register() {
+		$this->register_rest_field();
```

Comment:
No need for a bogus method call overhead here, you can just do the registration of the field directly within the `register()`, there's not much else to do anyway...
```suggestion
		register_rest_field(
			'user',
			self::USER_FIELD_DEVELOPER_TOOLS_ENABLED,
			[
				'get_callback'    => [ $this, 'rest_get_dev_tools_enabled' ],
				'update_callback' => [ $this, 'rest_update_dev_tools_enabled' ],
				'schema'          => [
					'description' => __( 'Whether the user has enabled dev tools.', 'amp' ),
					'type'        => 'boolean',
				],
			]
		);
```

Reply:
Yes, good point! Updated in https://github.com/ampproject/amp-wp/pull/4833/commits/71b8363bf8c394e99774ba16734c8c91ff363d15